aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -406,43 +406,60 @@ Tensor new_empty_strided(
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ eye ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Tensor eye(int64_t n,
-    c10::optional<ScalarType> dtype,
-    c10::optional<Layout> layout,
-    c10::optional<Device> device,
-    c10::optional<bool> pin_memory) {
-  // the default value of `m` equals to `n`
-  return native::eye(n, n, dtype, layout, device, pin_memory);
-}
-
-Tensor eye(int64_t n, int64_t m,
+    c10::optional<int64_t> m,
+    c10::optional<int64_t> k,
     c10::optional<ScalarType> dtype,
     c10::optional<Layout> layout,
     c10::optional<Device> device,
     c10::optional<bool> pin_memory) {
   // See [Note: hacky wrapper removal for TensorOptions]
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
 
+  // m defaults to n (square matrix)
+  // k defaults to 0 (main diagonal)
+  c10::optional<int64_t> m_v{n}, k_v{0};
+  if (m.has_value()){ m_v = m; }
+  if (k.has_value()) { k_v = k; }
   auto tensor = at::empty({0}, options); // to be resized
-  return at::eye_out(tensor, n, m);
+  return at::eye_out(tensor, n, m_v, k_v);
 }
 
-Tensor& eye_out_cpu(int64_t n, Tensor& result) {
-  // the default value of `m` equals to `n`
-  return native::eye_out_cpu(n, n, result);
-}
-
-Tensor& eye_out_cpu(int64_t n, int64_t m, Tensor& result) {
+Tensor& eye_out_cpu(int64_t n,
+                    c10::optional<int64_t> m,
+                    c10::optional<int64_t> k,
+                    Tensor& result) {
   TORCH_CHECK(n >= 0, "n must be greater or equal to 0, got ", n);
-  TORCH_CHECK(m >= 0, "m must be greater or equal to 0, got ", m);
+  TORCH_CHECK(m.value() >= 0, "m must be greater or equal to 0, got ", m.value());
```

Comment:
Need to first check that `m` and `k` actually have value, and if they dont, set them to their default value, right? Defining the arguments of the C++ funciton as `m_` or `k_` and then do
```c++
auto m = m_.has_value() ? m_.value() : n;
```
Then, you can do the `TORCH_CHECK`s as usual.

Reply:
I was wondering if it would be better to not assign to a new variable, but I see `TORCH_CHECK` isn't always the first thing in a function so, makes sense.